src/mono/mono/metadata/class.c

```diff
@@ -5445,21 +5445,70 @@ mono_field_get_rva (MonoClassField *field)
 
 	g_assert (field->type->attrs & FIELD_ATTRIBUTE_HAS_FIELD_RVA);
 
-	def_values = mono_class_get_field_def_values (klass);
+	def_values = mono_class_get_field_def_values_with_swizzle (klass, swizzle);
```

Comment:
This should special case swizzle=1 since thats the only one used in the little endian case.

Reply:
What would you like the special case to look like? The current logic puts the special case into `mono_class_get_field_def_values_with_swizzle` and `mono_class_set_field_def_values_with_swizzle`. Would you like to see an ifdef here for little endian which just ignores the swizzle parameter?

Reply:
Its ok, I tought that set_pointer_property () etc. is really slow, but its not.